artemis/frontend.py

```diff
@@ -122,18 +136,49 @@ async def post_add(
         )
 
 
+@router.get("/analysis/remove-pending-tasks/{analysis_id}", include_in_schema=False)
+def get_remove_pending_tasks(request: Request, analysis_id: str, csrf_protect: CsrfProtect = Depends()) -> Response:
+    return csrf.csrf_form_template_response(
+        "/remove_pending_tasks.jinja2",
+        {
+            "analysis_id": analysis_id,
+            "analysed_object": db.get_analysis_by_id(analysis_id)["payload"]["data"],  # type: ignore
+            "request": request,
+        },
+        csrf_protect,
+    )
+
+
+@router.post("/analysis/remove-pending-tasks/{analysis_id}", include_in_schema=False)
+@csrf.validate_csrf
+async def post_remove_pending_tasks(
+    request: Request, analysis_id: str, csrf_protect: CsrfProtect = Depends()
+) -> Response:
+    db.mark_analysis_as_stopped(analysis_id)
```

Comment:
I would swap places, firstly I would stop tasks in queue, then mark analysis as stopped

Reply:
why? That would be a race condition - If then after stopping tasks in queue an existing task would finish and add new tasks to queue, we would end up with non-finished analysis

Reply:
marking analysis as stopped prevents adding new tasks as well